storage/src/cache/filecache/cache_entry.rs

```diff
@@ -1058,8 +1070,12 @@ impl FileIoMergeState {
         tag: BlobIoTag,
         chunk: Option<BlobIoChunk>,
     ) -> Result<()> {
-        if self.regions.is_empty() || !self.joinable(region_type) {
+        if self.regions.is_empty()
+            || !self.joinable(region_type)
+            || self.status == FileIoMergeStateStatus::Committed
```

Comment:
So it plays with the merge state status on a per-region manner but the status itself is per state. So sometimes a merge state is committed, and sometimes it is open, depending on the sequence when the user-IO is seen in the list?

Reply:
Yes. It is decided by caller where chunk's type and if it hits local cache matter.
For user-IO, it can be pushed to current region(the last on within merge state), which will be dispatched to read data from local cache file. But for internal IO that also hits local cache, we must skip read data from cache for internal IO since it will consume data buffers provided from Rafs.